### PR TITLE
[docs] fix 404s and some formatting

### LINKED ISSUE #8845: [docs] links to docdb replication
The links to docdb replication in latest/explore/fault-tolerance/*.md need to be updated.

Currently they have links in the form of:
`that allows a [fault tolerance](../../../architecture/concepts/docdb/replication/) of 1`   ... which yield 404.
They should instead link to something like:
`architecture/docdb-replication/replication/`

### PR DESCRIPTION
**Pull request by polarweasel**
Fixes #8845 (among others)

Build preview is https://deploy-preview-8851--infallible-bardeen-164bc9.netlify.app/latest/

### CODE REVIEW

**Comment by polarweasel** [comment]
Hey @ddhodge can you take a quick look at this one? I forgot to tag you last week :)

File: docs/content/latest/faq/general.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ If you have a specific use case in mind, share it in our [Slack community](https
...
+[DocDB](../../architecture/docdb/), YugabyteDB's distributed document store common across all APIs, is built using a custom integration of Raft replication, distributed ACID transactions and the RocksDB storage engine. Specifically, DocDB enhances RocksDB by transforming it from a key-value store (with only primitive data types) to a document store (with complex data types). **Every key is stored as a separate document in DocDB, irrespective of the API responsible for managing the key.** DocDB’s [sharding](../../architecture/docdb-sharding/sharding/), [replication/fault-tolerance](../../architecture/docdb-replication/replication/) and [distributed ACID transactions](../../architecture/transactions/distributed-txns/) architecture are all based on the [Google Spanner design](https://research.google.com/archive/spanner-osdi2012.pdf) first published in 2012. [How We Built a High Performance Document Store on RocksDB?](https://blog.yugabyte.com/how-we-built-a-high-performance-document-store-on-rocksdb/) provides an in-depth look into DocDB.
```
Comment (ddhodge) [inline docs/content/latest/faq/general.md:110]: Further down at ## What about correctness testing? there is an extra \

**Review by ddhodge — COMMENTED** [review]
LGTM

File: docs/content/latest/explore/indexes-constraints/constraints.md [hunk 2]
```diff
@@ -250,7 +250,7 @@ For additional examples, see [Table with CHECK constraint](https://docs.yugabyte
 
 The `UNIQUE` constraint allows you to ensure that values stored in columns are unique across rows in a table. During inserting new rows or updating existing ones, the `UNIQUE` constraint checks if the value is already in the table, in which case the change is rejected and an error is displayed.
 
-When you add a `UNIQUE` constraint to one or more columns, YSQL automatically creates a [unique index](../indexes-1#using-the-unique-index) on these columns.
```
Comment (ddhodge) [inline docs/content/latest/explore/indexes-constraints/constraints.md:253]: Table with CHECK constraint link above is hard link

**Review by ddhodge — COMMENTED** [review]
LGTM minor link stuff

**Comment by polarweasel** [comment]
> Foreign key, Table with foreign key, and Table with CHECK constraint links above also use hard links 

And there were even more than that... Fixed them all!

File: docs/content/latest/explore/indexes-constraints/constraints.md [hunk 3]
```diff
@@ -176,14 +176,14 @@ ALTER TABLE child_table
...
   DROP CONSTRAINT constraint_fkey;
 ```
 
 The next step is to add a new foreign key constraint, possibly including an action, as demonstrated by the following syntax:
 
-```
+```sql
 ALTER TABLE child_table
   ADD CONSTRAINT constraint_fk
     FOREIGN KEY (fk_columns)
```
Comment (ddhodge) [inline docs/content/latest/explore/indexes-constraints/constraints.md:189]: Links below appear to be broken and hard links

File: docs/content/latest/explore/indexes-constraints/constraints.md [hunk 4]
```diff
@@ -191,10 +191,10 @@ ALTER TABLE child_table
       [ON DELETE action];
 ```
 
-For more information and examples, see the following: 
+For more information and examples, see the following:
 
-- [Foreign Key](./../../api/ysql/the-sql-language/statements/ddl_create_table/#foreign-key)
-- [Table with Foreign Key](./../../api/ysql/the-sql-language/statements/ddl_create_table/#table-with-foreign-key-constraint)
+- [Foreign Key](../../../api/ysql/the-sql-language/statements/ddl_create_table/#foreign-key)
+- [Table with Foreign Key](../../../api/ysql/the-sql-language/statements/ddl_create_table/#table-with-foreign-key-constraint)
```
Comment (ddhodge) [inline docs/content/latest/explore/indexes-constraints/constraints.md:197]: Same problem with Table with Check Constraint link below

File: docs/content/latest/explore/indexes-constraints/constraints.md [hunk 5]
```diff
@@ -244,13 +244,13 @@ ALTER TABLE employees
   ADD CONSTRAINT name_check CHECK (char_length(name) <= 3);
 ```
 
-For additional examples, see [Table with CHECK constraint](https://docs.yugabyte.com/latest/api/ysql/the-sql-language/statements/ddl_create_table/#table-with-check-constraint).
+For additional examples, see [Table with CHECK constraint](./../../api/ysql/the-sql-language/statements/ddl_create_table/#table-with-check-constraint).
```
Comment (polarweasel) [inline docs/content/latest/explore/indexes-constraints/constraints.md:247]:
```suggestion
For additional examples, see [Table with CHECK constraint](../../../api/ysql/the-sql-language/statements/ddl_create_table/#table-with-check-constraint).
```

**Pull request merged**